client/src/api/managment-service.ts

```diff
@@ -612,6 +663,131 @@ class ManagmentServiceApi {
     await this.refreshTokenIfExpired();
     await this.axiosInstance.put('/api/v1/clusters', clusterData);
   }
+
+  public async getSlackChannels(): Promise<SlackNotificationChannel[]> {
+    await this.refreshTokenIfExpired();
+    const slackChannels = await this.axiosInstance.get(
+      '/api/v1/notification-channels/slack',
+    );
+    return slackChannels.data;
+  }
+
+  public async getDiscordChannels(): Promise<DiscordNotificationChannel[]> {
+    await this.refreshTokenIfExpired();
+    const discordChannels = await this.axiosInstance.get(
+      '/api/v1/notification-channels/discord',
+    );
+    return discordChannels.data;
+  }
+
+  public async getEmailChannels(): Promise<EmailNotificationChannel[]> {
+    await this.refreshTokenIfExpired();
+    const emailChannels = await this.axiosInstance.get(
+      '/api/v1/notification-channels/mails',
+    );
+    return emailChannels.data;
+  }
+
+  public async postSlackChannel(slackChannel: SlackChannelForm): Promise<void> {
+    await this.refreshTokenIfExpired();
+    await this.axiosInstance.post(
+      '/api/v1/notification-channels/slack',
+      slackChannel,
+    );
+  }
+
+  public async postDiscordChannel(
+    discordChannel: DiscordChannelForm,
+  ): Promise<void> {
+    await this.refreshTokenIfExpired();
+    await this.axiosInstance.post(
+      '/api/v1/notification-channels/discord',
+      discordChannel,
+    );
+  }
+
+  public async postEmailChannel(emailChannel: EmailChannelForm): Promise<void> {
+    await this.refreshTokenIfExpired();
+    await this.axiosInstance.post(
+      '/api/v1/notification-channels/mails',
+      emailChannel,
+    );
+  }
+
+  public async editDiscordChannel(data: EditDiscordChannelForm): Promise<void> {
+    await this.refreshTokenIfExpired();
+    await this.axiosInstance.patch(
+      `/api/v1/notification-channels/discord/${data.id}`,
+      {
+        name: data.name,
+        webhookUrl: data.webhookUrl === '' ? undefined : data.webhookUrl,
```

Comment:
undefined as webhookUrl? Can we just prevent user from sending this field if it's empty?

Reply:
undefined means that this field will not be sent

Reply:
my lack of knowledge is visible 🫣